index.js

```diff
@@ -35,26 +40,44 @@ class BootSimulator {
   }
 
   async filterOnlyiOSSimulators() {
+    let device;
     list = await this.listSimulators();
     await _.map(list, (key, value) => {
       if (value.startsWith('iOS')) {
         _.forEach(key, values => {
+          if(values.name.replace(/ /g,'').toUpperCase().includes(deviceparam.replace(/ /g,'').toUpperCase()))
           this.deviceInfo.push(values.name + '-' + value);
         });
       }
     });
+    if(this.deviceInfo.length<1){
+      console.log("No device matches with provided device name");
+      device='';
+    }
+    else if(this.deviceInfo.length>1){
     const answer = await inquirer.prompt({
       type: 'list',
       name: 'Device',
       message: 'Select Simulator 📱 to boot?',
       pageSize: 15,
       choices: this.deviceInfo
     });
-    let device = await _.split(answer.Device, '-');
+    device = await _.split(answer.Device, '-');
+  }
+  else{
+    device= await _.split(this.deviceInfo[0],'-');
+
+  }
+
     let udid = await getAllSimulator.findUDIDOfSim(device[0], device[1]);
+    if(udid){
+    console.log("Booting "+device[0]+" with "+device[1]);
     await exec('xcrun simctl boot ' + udid);
+    await exec('open /Applications/Xcode.app/Contents/Developer/Applications/Simulator.app/');
```

Comment:
Instead try this `open -a Simulator --args -CurrentDeviceUDID CB194DE8-97DF-4DAB-BA06-CC15274E0B40` to launch simulator

Reply:
I thing open -a Simulator is enough to open the simulator but i think even this command expects the Xcode to be under /Applications. I'm not sure if this command will work if the Xcode is at custom path. @SrinivasanTarget